AmplifyPlugins/API/AWSAPICategoryPluginFunctionalTests/BlogPostCommentGraphQLWithAPIKey/BlogPostCommentGraphQLWithAPIKeyTests.swift

```diff
@@ -248,8 +249,8 @@ class BlogPostCommentGraphQLWithAPIKeyTests: XCTestCase {
         return blog
     }
 
-    func createPost(postBlogId: String, title: String) -> Post? {
-        var post: Post?
+    func createPost(postBlogId: String, title: String) -> AWSAPICategoryPluginTestCommon.Post? {
+        var post: AWSAPICategoryPluginTestCommon.Post?
```

Comment:
What is the difference between the two `Post` flavors in AmplifyTestCommon and AWSAPICategoryPluginTestCommon? Any opportunity to collapse those to reduce confusion? If not, consider renaming one of the models altogether so we're not surprised by unexpected divergence in behavior.

Reply:
One was hand crafted without using model gen to make it work with the document based GraphQL requests when building the API plugin and testing the default Blog/Post/Comment schema provided by amplify CLI, and the other is the model generated Post/Comment from the DataStore post/comment schema that was hand crafted.

i think the work i want to capture is to 
1. use the default blog/post/comment schema provided by amplify-cli for model gen (update schema in AmplifyTestCommon)
2. model gen and place the model files in AmplifyTestCommon
3. Remove the hand crafted types in AWSAPICategoryPluginTestCommon.
4. migrate the test cases in `BlogPostCommentGraphQLWithAPIKeyTests` over to GraphQLModelBasedTests. The tests are using the document based APIs, so we should take the models, and the GraphQLDocument helper classes to construct the GraphQLRequest for the document based APIs.
5. `GraphQLModelBasedTests` can probably renamed to something more specific like `BlogPostCommentAPITests`

this would show that using the amplify-cli schema works with model gen, how to call API, using the model based APIs and the document based APIs.

Reply:
since this PR is mostly to get storage up and running, including some of the refactoring of the common test helper classes over to core, will capture any API work in separate issue: https://github.com/aws-amplify/amplify-ios/issues/263